features/net/network-socket/nsapi_dns.cpp

```diff
@@ -261,20 +377,18 @@ static int nsapi_dns_query_multiple(NetworkStack *stack,
 }
 
 // convenience functions for other forms of queries
-extern "C" int nsapi_dns_query_multiple(nsapi_stack_t *stack,
-        nsapi_addr_t *addr, unsigned addr_count,
-        const char *host, nsapi_version_t version)
+extern "C" int nsapi_dns_query_multiple(nsapi_stack_t *stack, const char *host,
+        nsapi_addr_t *addr, unsigned addr_count, nsapi_version_t version)
 {
     NetworkStack *nstack = nsapi_create_stack(stack);
-    return nsapi_dns_query_multiple(nstack, addr, addr_count, host, version);
+    return nsapi_dns_query_multiple(nstack, host, addr, addr_count, version);
```

Comment:
It looks like this change and a lot of the changes below it are just changing the order of parameters. In the future it would be good to do this in a separate refactoring commit. That makes it easier to review both the refactor and the new feature.